### PR TITLE
feat(Core/Config): Configurable buy/sell rates

### PR DESCRIPTION
**Pull request by Darkon47**
## CHANGES PROPOSED:
-  Add Rates for item sell value and item buy value based on item quality, and apply them as template changes when the item templates are loaded

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2743

## TESTS PERFORMED:
Untested

## KNOWN ISSUES AND TODO LIST:
-Add Conf options for Drop amounts for profession uses [ ]
-Add Conf options for game object uses for profession gathering nodes [ ] 

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/Globals/ObjectMgr.cpp [hunk 1]
```diff
@@ -2314,8 +2314,8 @@ void ObjectMgr::LoadItemTemplates()
         itemTemplate.Flags                     = fields[7].GetUInt32();
         itemTemplate.Flags2                    = fields[8].GetUInt32();
         itemTemplate.BuyCount                  = uint32(fields[9].GetUInt8());
-        itemTemplate.BuyPrice                  = int32(fields[10].GetInt64());
-        itemTemplate.SellPrice                 = fields[11].GetUInt32();
+        itemTemplate.BuyPrice                  = int32(fields[10].GetInt64()*sWorld->getRate((Rates)(RATE_BUYVALUE_ITEM_POOR + itemTemplate.Quality));
+        itemTemplate.SellPrice                 = fields[11].GetUInt32()*sWorld->getRate((Rates)(RATE_SELLVALUE_ITEM_POOR+itemTemplate.Quality));
```
Comment (BarbzYHOOL) [inline src/server/game/Globals/ObjectMgr.cpp]:
```suggestion
        itemTemplate.SellPrice                 = fields[11].GetUInt32()*sWorld->getRate((Rates)(RATE_SELLVALUE_ITEM_POOR + itemTemplate.Quality));
```

**Comment by pklloveyou** [comment]
Tested, good

**Comment by pklloveyou** [comment]
Very useful features

File: src/server/game/Globals/ObjectMgr.cpp [hunk 2]
```diff
@@ -2314,8 +2314,8 @@ void ObjectMgr::LoadItemTemplates()
         itemTemplate.Flags                     = fields[7].GetUInt32();
         itemTemplate.Flags2                    = fields[8].GetUInt32();
         itemTemplate.BuyCount                  = uint32(fields[9].GetUInt8());
-        itemTemplate.BuyPrice                  = int32(fields[10].GetInt64());
-        itemTemplate.SellPrice                 = fields[11].GetUInt32();
+        itemTemplate.BuyPrice                  = int32(fields[10].GetInt64()*sWorld->getRate((Rates)(RATE_BUYVALUE_ITEM_POOR + itemTemplate.Quality)));
+        itemTemplate.SellPrice                 = uint32(fields[11].GetUInt32()*sWorld->getRate((Rates)(RATE_SELLVALUE_ITEM_POOR+itemTemplate.Quality)));
```
Comment (FrancescoBorzi) [inline src/server/game/Globals/ObjectMgr.cpp]: please always put spaces around operators like `*` and `+`

**Comment by FrancescoBorzi** [comment]
@Darkon47 another tip: NEVER use the  `master` branch of your fork when creating a PR, but create a new branch per every PR instead.

http://www.azerothcore.org/wiki/How-to-create-a-PR

Overal looks good, well done!

**Pull request merged**